spec/lib/gamestate_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rspec'
+require 'rspec/core'
+require_relative '../../lib/gamestate.rb'
+
+describe Gamestate do
+  let(:castle_data) do [{"name"=>"Old Timey Medieval Castle",
+                         "rooms"=>[{
+                                   "name"=>"dungeon",
+                                       "monster"=>{"name"=>"skeleton warden", "win_chance"=>95},
+                                       "treasure"=>{"type"=>"jail cell key", "points"=>10}},
+                                   {"name"=>"the endless staircase",
+                                      "monster"=>{"name"=>"cursed princess", "win_chance"=>80},
+                                      "treasure"=>{"type"=>"a ruby necklace", "points"=>25}},
+                                   {"name"=>"throne room",
+                                      "monster"=>{"name"=>"the evil dragon", "win_chance"=>70},
+                                      "treasure"=>{"type"=>"the dragon's hoard", "points"=>100}}]}]
+  end
```

Comment:
This doesn't need to be a let variable. Do it in the gamestate let variable block.